Algo/src/roots.cxx

```diff
@@ -28,9 +29,11 @@ template <typename T> std::pair<vec<T, 2>, mat<T, 2> > gnd(vec<T, 2> x) {
 extern "C" void Test_roots(CCTK_ARGUMENTS) {
   DECLARE_CCTK_ARGUMENTS;
 
-  auto fn = [](auto x) { return x * x - 2; };
-  auto fnd = [](auto x) { return std::make_tuple(x * x - 2, 2 * x); };
-  auto fnd2 = [](auto x) { return std::make_tuple(x * x - 2, 2 * x, 2); };
+  auto fn = []CCTK_HOST CCTK_DEVICE (double x) { return x * x - 2; };
```

Comment:
Can you use `auto` (or `CCTK_REAL`) instead of `double`?

Reply:
should I change all the `double` to `CCTK_REAL` which appears in the file